Unable to build the library

Hi, I just cloned the lib and tried to `npm run build`, I get the following error:

![error-ng2-notifications](https://cloud.githubusercontent.com/assets/570905/15778511/ba5f2634-29a7-11e6-84ce-c6d27a9e7820.png)

> λ npm run build
> 
> angular2-notifications@0.1.56 build C:\Users\Abdel\Desktop\angular2-notifications
> tsc
> 
> node_modules/@angular/core/src/application_ref.d.ts(39,88): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/application_ref.d.ts(99,42): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/application_ref.d.ts(174,33): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/change_detection/differs/default_keyvalue_differ.d.ts(24,15): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/change_detection/differs/default_keyvalue_differ.d.ts(26,16): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/di/reflective_provider.d.ts(105,123): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/di/reflective_provider.d.ts(105,165): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/async.d.ts(27,33): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/async.d.ts(28,45): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/collection.d.ts(1,25): error TS2304: Cannot find name 'MapConstructor'.
> node_modules/@angular/core/src/facade/collection.d.ts(2,25): error TS2304: Cannot find name 'SetConstructor'.
> node_modules/@angular/core/src/facade/collection.d.ts(4,27): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/collection.d.ts(4,39): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/collection.d.ts(7,9): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/collection.d.ts(8,30): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/collection.d.ts(11,43): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/collection.d.ts(12,27): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/collection.d.ts(14,23): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/collection.d.ts(15,25): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/collection.d.ts(100,41): error TS2304: Cannot find name 'Set'.
> node_modules/@angular/core/src/facade/collection.d.ts(101,22): error TS2304: Cannot find name 'Set'.
> node_modules/@angular/core/src/facade/collection.d.ts(102,25): error TS2304: Cannot find name 'Set'.
> node_modules/@angular/core/src/facade/lang.d.ts(4,17): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/lang.d.ts(5,17): error TS2304: Cannot find name 'Set'.
> node_modules/@angular/core/src/facade/lang.d.ts(70,59): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/facade/promise.d.ts(2,14): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/promise.d.ts(8,32): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/promise.d.ts(9,38): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/promise.d.ts(10,35): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/promise.d.ts(10,93): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/promise.d.ts(11,34): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/promise.d.ts(11,50): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/promise.d.ts(12,32): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/promise.d.ts(12,149): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/facade/promise.d.ts(13,43): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/linker/component_resolver.d.ts(8,53): error TS2304: Cannot find name 'Promise'. node_modules/@angular/core/src/linker/component_resolver.d.ts(12,44): error TS2304: Cannot find name 'Promise'.
> 
> node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(62,148): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(103,144): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(108,139): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(109,135): error TS2304: Cannot find name 'Promise'.
> node_modules/@angular/core/src/reflection/reflector.d.ts(28,22): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/reflection/reflector.d.ts(30,15): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/reflection/reflector.d.ts(32,15): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/reflection/reflector.d.ts(34,15): error TS2304: Cannot find name 'Map'.
> node_modules/@angular/core/src/reflection/reflector.d.ts(36,16): error TS2304: Cannot find name 'Set'.
> node_modules/@angular/core/src/testability/testability.d.ts(40,20): error TS2304: Cannot find name 'Map'.
> node_modules/rxjs/Observable.d.ts(10,66): error TS2304: Cannot find name 'Promise'.
> node_modules/rxjs/Observable.d.ts(66,60): error TS2304: Cannot find name 'Promise'.
> node_modules/rxjs/Observable.d.ts(66,70): error TS2304: Cannot find name 'Promise'.
> src/pushNotifications.service.ts(10,18): error TS4073: Parameter 'data' of public method from exported class has or is using private name 'PushNotification'.
> 
> npm ERR! Windows_NT 10.0.10586
> npm ERR! argv "D:\NodeJS\node.exe" "D:\NodeJS\node_modules\npm\bin\npm-cli.js" "run" "build"
> npm ERR! node v4.2.1
> npm ERR! npm  v3.9.5
> npm ERR! code ELIFECYCLE
> npm ERR! angular2-notifications@0.1.56 build: `tsc`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the angular2-notifications@0.1.56 build script 'tsc'.
> npm ERR! Make sure you have the latest version of node.js and npm installed.
> npm ERR! If you do, this is most likely a problem with the angular2-notifications package,
> npm ERR! not with npm itself.
> npm ERR! Tell the author that this fails on your system:
> npm ERR!     tsc
> npm ERR! You can get information on how to open an issue for this project with:
> npm ERR!     npm bugs angular2-notifications
> npm ERR! Or if that isn't available, you can get their info via:
> npm ERR!     npm owner ls angular2-notifications
> npm ERR! There is likely additional logging output above.

Any clue what I am doing wrong?

--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,6 @@
 npm-debug.log
 node_modules
+typings
 example/node_modules
 example/app/**/**.js
 .idea

--- a/package.json
+++ b/package.json
@@ -3,7 +3,7 @@
   "version": "0.1.56",
   "description": "An easy to use notification library for angular 2",
   "scripts": {
-    "build": "tsc"
+    "build": "typings install && tsc"
   },
   "repository": {
     "type": "git",
@@ -27,16 +27,16 @@
     "@angular/core": "^2.0.0-rc.1",
     "rxjs": "5.0.0-beta.6"
   },
-  "main": "./components.js",
   "devDependencies": {
-    "gulp": "^3.9.1",
-    "typescript": "^1.8.10",
-    "typings": "^0.7.12",
     "@angular/core": "^2.0.0-rc.1",
     "es6-shim": "^0.35.0",
+    "gulp": "^3.9.1",
     "reflect-metadata": "0.1.2",
     "rxjs": "5.0.0-beta.6",
+    "typescript": "^1.8.10",
+    "typings": "^0.7.12",
     "zone.js": "~0.6.12"
   },
+  "main": "./components.js",
   "typings": "./components.d.ts"
 }

--- a/src/pushNotifications.service.ts
+++ b/src/pushNotifications.service.ts
@@ -8,7 +8,7 @@ export class PushNotificationsService {
     private notificationBuffer: PushNotification;
 
     create(data: PushNotification): any {
-        
+
         if (!this.canCreate) {
             this.notificationBuffer = data;
             this.getPermission();
@@ -49,7 +49,7 @@ export class PushNotificationsService {
     }
 }
 
-interface PushNotification {
+export interface PushNotification {
     title: string
     body: string
 }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -11,6 +11,8 @@
     "declaration": true
   },
   "files": [
+    "typings/browser.d.ts",
+    "components.d.ts",
     "src/icons.ts",
     "src/max.pipe.ts",
     "src/notification.component.ts",

--- a/typings.json
+++ b/typings.json
@@ -1,0 +1,5 @@
+{
+  "ambientDependencies": {
+    "es6-shim": "registry:dt/es6-shim#0.31.2+20160602141504"
+  }
+}

